db/python/layers/family.py

```diff
@@ -232,12 +233,33 @@ class FamilyLayer(BaseLayer):
 
     def __init__(self, connection: Connection):
         super().__init__(connection)
+        self.stable = SampleTable(connection)
         self.ftable = FamilyTable(connection)
         self.fptable = FamilyParticipantTable(self.connection)
 
-    async def get_families(self, project: int = None):
+    async def get_families(
+        self,
+        project: int = None,
+        participant_ids: List[int] = None,
+        sample_ids: List[int] = None,
+    ):
         """Get all families for a project"""
-        return await self.ftable.get_families(project=project)
+        project = project or self.connection.project
+
+        # Merge sample_id and participant_ids into a single list
+        all_participants = participant_ids if participant_ids else []
+
+        # Find the participants from the given samples
+        _, samples = await self.stable.get_samples_by(
+            project_ids=[project], sample_ids=sample_ids
+        )
+
+        all_participants += [int(s.participant_id) for s in samples if s.participant_id]
+        all_participants = list(set(all_participants))
```

Comment:
When samples is false-y (empty list / unspecified), this returns all participant IDs within the project (that have samples) - regardless of the specified participants.

Reply:
This is interesting, because it's kind of the initial expected behaviour. i.e., return all families, but it does kind of seem unintuitive. Would you imagine this actually being split into a separate endpoint? e.g. `get_all_families` and `get_families_by_participants`?